internal/jobworker/cgroup/service.go

```diff
@@ -0,0 +1,239 @@
+// Package cgroup provides types for interaction with Linux cgroups v2.
+package cgroup
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/tjper/teleport/internal/log"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+// logger is an object for logging package events to stdout.
+var logger = log.New(os.Stdout, "cgroups")
+
+// NewService creates a Service instance.
+func NewService() (*Service, error) {
+	s := &Service{
+		// path has a random ID to ensure Service instance bases do not collide.
+		path: path.Join(mountPath, jobWorkerBase),
+	}
+
+	if err := s.mount(); err != nil {
+		return nil, err
+	}
+
+	controllers := []string{
+		cpu,
+		memory,
+		io,
+	}
+	if err := s.enableControllers(mountPath, controllers); err != nil {
+		return nil, err
+	}
+	if err := s.enableControllers(s.path, controllers); err != nil {
+		return nil, err
+	}
```

Comment:
Enabling and disabling Cgroup controllers is handled by `controller` interface. Enabling the controllers here is only for the /cgroup2 and /cgroup2/jobworker Cgroups. I do think this could a be a bit cleaner though. I'm going to experiment with walking the directory quick and see how it reads.

Reply:
Walking the directory seemed complex compared to enabling the controllers in the two cgroups they needed be enabled in. I've moved these calls into a single `s.enableControllers` to increase readability.

Reply:
Hmm, If I use the TempDir approach you outlined earlier, this is necessary. I'm going to work through it a bit more.